src/app/profiles/profile-detail/profile-detail.component.ts

```diff
@@ -123,37 +107,46 @@ export class ProfileDetailComponent implements OnInit {
     })
   }
 
-  generateRandomPasswordChange (value: boolean): void {
-    if (value) {
-      this.profileForm.controls.amtPassword.disable()
-      this.profileForm.controls.amtPassword.setValue(null)
-      this.profileForm.controls.amtPassword.clearValidators()
-      this.profileForm.controls.amtPassword.setValue(null)
-      this.profileForm.controls.passwordLength.setValue(8)
-      this.profileForm.controls.passwordLength.enable()
-    } else {
-      this.profileForm.controls.amtPassword.enable()
-      this.profileForm.controls.amtPassword.setValidators(Validators.required)
-      this.profileForm.controls.passwordLength.disable()
-      this.profileForm.controls.passwordLength.setValue(null)
-      this.profileForm.controls.passwordLength.clearValidators()
+  generateRandomPassword (controlName: string, length: number): void {
```

Comment:
love it, but we need to make sure this is unit tested

Reply:
Yep